packages/stream-api/src/streamApi.ts

```diff
@@ -305,6 +371,28 @@ export class StreamApi {
     payload: UpdateCrossSectioningPayload,
     withResponse = true
   ): Promise<vertexvis.protobuf.stream.IStreamResponse> {
+    // If a section plane is provided, verify it is valid
+    if (payload.crossSectioning?.sectionPlanes != null) {
+      let invalidSectionPlane = false;
+      for (const plane of payload.crossSectioning.sectionPlanes) {
+        const validNormal =
+          plane.normal != null && validateVector(plane.normal);
+        const validOffset =
+          plane.offset != null && validateNumber(plane.offset);
+
+        if (!validNormal || !validOffset) {
+          invalidSectionPlane = true;
+        }
+      }
+
+      if (invalidSectionPlane) {
+        console.warn(
+          'Invalid cross section plane provided. Canceling updateCrossSectioning operation.'
+        );
+        return Promise.resolve({});
+      }
+    }
+
```

Comment:
Nit: it looks like this could potentially be simplified a bit using `Array.some`:
```typescript
const invalidSectionPlane = payload.crossSectioning?.sectionPlanes?.some(
  (plane) => {
    const validNormal =
      plane.normal != null && validateVector(plane.normal);
    const validOffset =
      plane.offset != null && validateNumber(plane.offset);

    return !validNormal || !validOffset;
  }
);

if (invalidSectionPlane) {
  console.warn(
    'Invalid cross section plane provided. Canceling updateCrossSectioning operation.'
  );
  return Promise.resolve({});
}
```

Reply:
Good idea - updated!